Source/Core/VideoCommon/XFStructs.cpp

```diff
@@ -300,3 +300,511 @@ void PreprocessIndexedXF(u32 val, int refarray)
+    return DescriptionlessReg(XFMEM_SETGPMETRIC);
+
+  case XFMEM_CLIPDISABLE:
+    return std::make_pair(
+        RegName(XFMEM_CLIPDISABLE),
+        fmt::format("{} clipping detection, {} trivial rejection, {} cpoly clipping acceleration",
+                    (value & 1) ? "Enable" : "Disable", (value & 2) ? "enable" : "disable",
+                    (value & 3) ? "enable" : "disable"));
+
+  case XFMEM_VTXSPECS:  //__GXXfVtxSpecs, wrote 0004
+    return std::make_pair(RegName(XFMEM_VTXSPECS),
+                          fmt::format("Vertex specs: {}", INVTXSPEC{.hex = value}));
+
+  case XFMEM_SETNUMCHAN:
+    return std::make_pair(RegName(XFMEM_SETNUMCHAN),
+                          fmt::format("Number of color channels: {}", value & 3));
+    break;
+
+  case XFMEM_SETCHAN0_AMBCOLOR:  // Channel Ambient Color
+    return std::make_pair(RegName(XFMEM_SETCHAN0_AMBCOLOR),
+                          fmt::format("Channel 0 Ambient Color: {:x}", value));
+  case XFMEM_SETCHAN1_AMBCOLOR:
+    return std::make_pair(RegName(XFMEM_SETCHAN1_AMBCOLOR),
+                          fmt::format("Channel 1 Ambient Color: {:x}", value));
+
+  case XFMEM_SETCHAN0_MATCOLOR:  // Channel Material Color
+    return std::make_pair(RegName(XFMEM_SETCHAN0_MATCOLOR),
+                          fmt::format("Channel 0 Material Color: {:x}", value));
+  case XFMEM_SETCHAN1_MATCOLOR:
+    return std::make_pair(RegName(XFMEM_SETCHAN1_MATCOLOR),
+                          fmt::format("Channel 1 Material Color: {:x}", value));
+
+  case XFMEM_SETCHAN0_COLOR:  // Channel Color
+    return std::make_pair(RegName(XFMEM_SETCHAN0_COLOR),
+                          fmt::format("Channel 0 Color: {}", LitChannel{.hex = value}));
+  case XFMEM_SETCHAN1_COLOR:
+    return std::make_pair(RegName(XFMEM_SETCHAN1_COLOR),
+                          fmt::format("Channel 1 Color: {}", LitChannel{.hex = value}));
+  case XFMEM_SETCHAN0_ALPHA:  // Channel Alpha
+    return std::make_pair(RegName(XFMEM_SETCHAN0_ALPHA),
+                          fmt::format("Channel 0 Alpha: {}", LitChannel{.hex = value}));
+  case XFMEM_SETCHAN1_ALPHA:
+    return std::make_pair(RegName(XFMEM_SETCHAN1_ALPHA),
+                          fmt::format("Channel 1 Alpha: {}", LitChannel{.hex = value}));
+
+  case XFMEM_DUALTEX:
+    return std::make_pair(RegName(XFMEM_DUALTEX),
+                          fmt::format("Dual Tex Trans {}", (value & 1) ? "enabled" : "disabled"));
+
+  case XFMEM_SETMATRIXINDA:
+    return std::make_pair(RegName(XFMEM_SETMATRIXINDA),
+                          fmt::format("Matrix index A: {}", TMatrixIndexA{.Hex = value}));
+  case XFMEM_SETMATRIXINDB:
+    return std::make_pair(RegName(XFMEM_SETMATRIXINDB),
+                          fmt::format("Matrix index B: {}", TMatrixIndexB{.Hex = value}));
+
+  case XFMEM_SETVIEWPORT:
+    return std::make_pair(RegName(XFMEM_SETVIEWPORT + 0),
+                          fmt::format("Viewport width: {}", reinterpret_cast<float&>(value)));
```

Comment:
Consider using BitCast within Common/BitUtils for conversions like this. This invokes undefined behavior as is. Ditto for other occurrences below.